### PR TITLE
Configure strict remote data checking

### LINKED ISSUE #18: Plugin blocks internet access for all tests not using decorator
Currently, if this plugin is installed it will block internet access to **all** tests that are not marked with ``@pytest.mark.remote_data``. This is the desired behavior for `astropy` and some affiliated packages. However, just having this plugin installed will affect other packages that don't care about the ``remote_data`` plugin.

It seems like the best solution is to have a configuration option to enable the strict behavior. Packages that wish to enable this behavior will add a field to their `setup.cfg` file. Packages that don't care (and therefore don't have this option in their `setup.cfg` file) won't be affected.

### PR DESCRIPTION
**Pull request by drdavella**
This fixes #18 by allowing strict remote data checking to be configurable. This means that tests that access remote data sources but are not marked with `@pytest.mark.remote_data` will only fail if the configuration option `remote_data_strict` has been set in `setup.cfg`.

Astropy and affiliated packages that want this behavior will have to add `remote_data_strict = true` to their `setup.cfg` files.

### CODE REVIEW

File: pytest_remotedata/plugin.py [hunk 1]
```diff
@@ -16,21 +16,29 @@ def pytest_addoption(parser):
         "--remote-data", nargs="?", const='any', default='none',
         help="run tests with online data")
 
+    parser.addini('remote_data_strict',
+        "If 'True', tests will fail if they attempt to access the internet "
+        "internet but are not explicitly marked with 'remote_data'",
```
Comment (drdavella) [inline pytest_remotedata/plugin.py]: Documentation bug

**Comment by pllim** [comment]
I was just wondering if strict should be default, but rather have an option to opt out. So in that case, astropy don't need to add extra config, but JWST pipeline does.

**Comment by pllim** [comment]
And in the case of JWST pipeline, where its usage is actually defined within `asdf` dependency, how would this feature affect the dynamics over there? That is, would opting out have unintended consequences for the `asdf` dependency part?

**Comment by drdavella** [comment]
@pllim no I think the default needs to be disabled. Otherwise this plugin could potentially cause failures for lots of other packages that care nothing about remote data tests.

**Comment by drdavella** [comment]
And wrt asdf/jwst, the option is configured by the package being tested, so even if a dependency has this option set in it's own `setup.cfg`, it should not affect the package being tested if it does not.

**Comment by pllim** [comment]
@drdavella , that is a good point. But for packages that already using `remote_data`, now those marked tests won't be skipped anymore unless they add this "strict" config, is that right?

**Comment by bsipocz** [comment]
I was also thinking about packages that already use `remote-data`. Is that right that they cannot switch to use it from this package while they keep up support for astropy <3.0, as the plugins there and here are conflicting?

**Comment by drdavella** [comment]
@bsipocz, yes that is true right now. I have started thinking about whether there's something we can backport to v2.0.x to make it work, but I'm not sure whether it will be possible.

**Comment by drdavella** [comment]
@pllim, no this option only affects whether tests that access the internet but are **not** marked with `@pytest.mark.remote_data` fail. The rest of the `remote_data` functionality should work as expected.

**Comment by bsipocz** [comment]
Well, if we could do some backporting it would be brilliant.

**Pull request merged**